framework/include/physics/PhysicsBase.h

```diff
@@ -233,6 +331,122 @@ PhysicsBase::checkVectorParamsSameLength(const std::string & param1,
+    const auto size_2 = getParam<MultiMooseEnum>(param2).size();
+    if (size_1 != size_2)
+      paramError(param1,
+                 "Vector parameters '" + param1 + "' (size " + std::to_string(size_1) + ") and '" +
+                     param2 + "' (size " + std::to_string(size_2) + ") must be the same size");
+  }
+  // handle empty vector defaults
+  else if (isParamValid(param1) || isParamValid(param2))
+    if (getParam<std::vector<T>>(param1).size() || getParam<MultiMooseEnum>(param2).size())
+      checkParamsBothSetOrNotSet(param1, param2);
+}
+
+template <typename T, typename S>
+void
+PhysicsBase::checkTwoDVectorParamsSameLength(const std::string & param1,
+                                             const std::string & param2) const
+{
+  checkVectorParamsSameLength<std::vector<T>, std::vector<S>>(param1, param2);
+  if (isParamValid(param1) && isParamValid(param2))
+  {
+    const auto value1 = getParam<std::vector<std::vector<T>>>(param1);
+    const auto value2 = getParam<std::vector<std::vector<S>>>(param2);
+    for (const auto index : index_range(value1))
+      if (value1[index].size() != value2[index].size())
+        paramError(param1,
+                   "Vector at index " + std::to_string(index) + " of 2D vector parameter '" +
+                       param1 +
+                       "' is not the same size as its counterpart from 2D vector parameter '" +
+                       param2 + "'");
+  }
+  // handle empty vector defaults
+  else if (isParamValid(param1) || isParamValid(param2))
+    if (getParam<std::vector<T>>(param1).size() || getParam<std::vector<T>>(param2).size())
+      checkParamsBothSetOrNotSet(param1, param2);
+}
+
+template <typename T, typename S>
+void
+PhysicsBase::checkTwoDVectorParamInnerSameLengthAsOneDVector(const std::string & param1,
+                                                             const std::string & param2) const
+{
+  assertParamDefined<std::vector<std::vector<T>>>(param1);
+  assertParamDefined<std::vector<S>>(param2);
+  for (const auto & sub_vec_i : index_range(getParam<std::vector<std::vector<T>>>(param1)))
+  {
+    const auto size_1 = getParam<std::vector<std::vector<T>>>(param1)[sub_vec_i].size();
+    const auto size_2 = getParam<std::vector<S>>(param2).size();
+    if (size_1 != size_2)
+      paramError(param1,
+                 "Vector at index " + std::to_string(sub_vec_i) + " (size " +
+                     std::to_string(size_1) +
+                     ") "
+                     " of this parameter should be the same length as parameter '" +
+                     param2 + "' (size " + std::to_string(size_2) + ")");
+  }
+}
+
+template <typename T>
+void
+PhysicsBase::checkTwoDVectorParamMultiMooseEnumSameLength(const std::string & param1,
+                                                          const std::string & param2,
+                                                          const bool error_for_param2) const
```

Comment:
sometimes, saying :
param1 is not the same size as param2 is clearer

because for example the number of inlet boundaries not being the same as the number of inlet functors points to the inlet when the functors are probably the problem